node/utils/urlpattern.ts

```diff
@@ -0,0 +1,7 @@
+import { URLPattern } from 'urlpattern-polyfill'
+
+export class ExtendedURLPattern extends URLPattern {
+  // @ts-expect-error Internal property that the underlying class is using but
+  // not exposing.
+  regexp: Record<string, RegExp>
```

Comment:
I don't think this is a good idea. This will most probably not be available once this is available in Node.js natively. Also looking at the docs of how `URLPattern` is supposed to be used, I do not think it matches our usecase of converting a glob to a regex.

Reply:
The mdn docs mention this library: https://github.com/pillarjs/path-to-regexp maybe we should use this?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/URL_Pattern_API

Reply:
Okay, up to you. I personally feel this is a terrible idea to rely on the internals and implementation details of other packages.

Reply:
I share the sentiment of this feeling "bad", but I think it's an example of us knowing the rules enough that we're allowed to break them. Since we're using a polyfill, the underlying implementation won't depend on the runtime, and since we have tests we'll notice when a change in the polyfill breaks this. I think it's fine - and if we find problems, as a last resort we can still reimplement the spec ourselves.

Reply:
> At the very least we should pin an exact version

Totally. I actually meant to do this but ended up forgetting about it. Done in https://github.com/netlify/edge-bundler/pull/392/commits/b308434058e725cef3be1a63377fed508a46237e.

> My concern here is that we're opening ourselves to the possibility of the internals of the polyfill changing, as it's not a public API

If the internals change and the property we're using disappears or gets renamed, our own tests would catch it.

I'm not opposed to using our own fork if this makes people more comfortable — in fact, we do already have [a fork](https://github.com/netlify/urlpattern-polyfill), which I created in order to [push a change upstream](https://github.com/kenchris/urlpattern-polyfill/pull/111).

But if we go down that path, I would rather keep it on its own repo from where we can easily merge upstream changes, run the tests, etc. Not sure what we get by vendoring it in with Edge Bundler?